doc/assessment.md

```diff
@@ -77,35 +106,133 @@ An assessment is broken down in to a list of zones, like this:
 
-PrairieLearn distinguishes between *assessments* and *assessment instances*. A *assessment* is determined by the code in an `assessments` directory, and is something like "Midterm 1". Given an assessment, PrairieLearn needs to generate the random set of questions and question variants for each student, and it is this selection that is the *assessment instance* for the student. There is only one copy of each assessment, but every student has their own assessment instance. The rules for updating assessment instances differ between `Homework` and `Exam` assessments.
+2. For each zone, concatenate the *slot questions* from each slot to form the total list of *available zone questions*. If `numberChoose` is specified for the zone, randomly select that many questions from the available zone questions to give the *zone questions* (defaults to selecting all of the available zone questions). For Exams, randomly shuffle the order of the *zone questions*.
 
-**`Exam` assessment updates:** Exam assessment instances are generated when the student starts the exam, and they are never automatically deleted, regenerated, or updated, even when the original assessment is changed in some way. This is a safety mechanism to avoid having students' assessments changed during an exam. However, if you want to force the regeneration of assessment instances then you can do so with the “reset” button on instructor view of the assessment. While writing an assessment you might need to do this many times. Once an assessment is live, you should of course be very careful about doing this (basically, don’t do it on a production server once an assessment is underway).
+3. Concatenate the *zone questions* from each zone to form the total set of *assessment questions* for the student. This set of questions forms the *assessment instance* for this student.
 
-**`Homework` assessment updates:** New questions added to Homeworks will be automatically integrated into student homeworks currently in progress. Updates to `maxPoints` will take effect the next time a student grades a question. A student's “points” and “percentage score” will never decrease.
+Each zone appears in the given order in the assessment (this is not randomized for each student). Within each zone the question order is randomized per-student for Exams, but not randomized for Homeworks (but see the `shuffleQuestions` option).
 
-## Multiple-instance versus single-instance assessments
+## Computing the points for each question
+
+FIXME
+
+## Computing the total points for an assessment
+
+An assessment consists of several zones, each of which contains several questions. To compute the total assessment points, we proceed in three steps:
```

Comment:
You may also want to mention here that "extra" credit can be awarded based on access rules?